File: backend/app/schemas/alert.py
"""Alert request/response schemas."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class AlertResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    alert_type: str
    severity: str
    message: str
    details: dict | None = None
    is_read: bool
    ingredient_id: int | None = None
    recipe_id: int | None = None
    created_at: datetime


class AlertListResponse(BaseModel):
    items: list[AlertResponse]
    total: int
    unread_count: int


class AlertCountResponse(BaseModel):
    unread_count: int

File: frontend/src/types/alert.ts
// Mirror of backend/app/schemas/alert.py

export interface AlertItem {
  id: number;
  alert_type: string;
  severity: string;
  message: string;
  details: Record<string, unknown> | null;
  is_read: boolean;
  ingredient_id: number | null;
  recipe_id: number | null;
  created_at: string;
}

export interface AlertListResponse {
  items: AlertItem[];
  total: number;
  unread_count: number;
}

export interface AlertCountResponse {
  unread_count: number;
}
